src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -174,6 +174,14 @@ namespace AppInstaller::CLI::Workflow
 
         // Overwrite the source with the composite.
         context.Add<Execution::Data::Source>(std::move(compositeSource));
+
+        if (m_predefinedSource == PredefinedSource::ARP_System ||
+            m_predefinedSource == PredefinedSource::ARP_User ||
+            m_predefinedSource == PredefinedSource::MSIX ||
+            m_predefinedSource == PredefinedSource::Installed)
+        {
+            context.SetFlags(Execution::ContextFlag::CompositeInstalledSource);
+        }

```

Comment:
Why not `ISource::IsComposite`?

Reply:
searching across multiple sources like "winget install foo" when there are more than 2 sources configured will make ISource::IsComposite() to be true, but we want installed composite source for our case

---
In reply to: [518336942](https://github.com/microsoft/winget-cli/pull/634#discussion_r518336942) [](ancestors = 518336942)

Reply:
Removed the flag after discussion

---
In reply to: [518454148](https://github.com/microsoft/winget-cli/pull/634#discussion_r518454148) [](ancestors = 518454148,518336942)